components/dada-execute/src/step/apply_op.rs

```diff
@@ -100,22 +106,34 @@ impl Stepper<'_> {
             },
             (&ObjectData::Float(lhs), &ObjectData::Float(rhs)) => match op {
                 Op::EqualEqual => Ok(self.machine.our_value(lhs == rhs)),
+                Op::GreaterEqual => Ok(self.machine.our_value(lhs >= rhs)),
+                Op::LessEqual => Ok(self.machine.our_value(lhs <= rhs)),
                 Op::Plus => Ok(self.machine.our_value(lhs + rhs)),
                 Op::Minus => Ok(self.machine.our_value(lhs - rhs)),
                 Op::Times => Ok(self.machine.our_value(lhs * rhs)),
                 Op::DividedBy => Ok(self.machine.our_value(lhs / rhs)),
                 Op::LessThan => Ok(self.machine.our_value(lhs < rhs)),
                 Op::GreaterThan => Ok(self.machine.our_value(lhs > rhs)),
             },
-            (ObjectData::String(lhs), ObjectData::String(rhs)) => {
-                let eq = lhs == rhs;
-                match op {
-                    Op::EqualEqual => Ok(self.machine.our_value(eq)),
-                    _ => op_error(),
+            (ObjectData::String(lhs), ObjectData::String(rhs)) => match op {
+                Op::EqualEqual => {
+                    let val = lhs == rhs;
+                    Ok(self.machine.our_value(val))
                 }
-            }
+                Op::GreaterEqual => {
+                    let val = lhs >= rhs;
+                    Ok(self.machine.our_value(val))
+                }
+                Op::LessEqual => {
+                    let val = lhs <= rhs;
+                    Ok(self.machine.our_value(val))
+                }
+                _ => op_error(),
+            },
             (&ObjectData::Unit(()), &ObjectData::Unit(())) => match op {
                 Op::EqualEqual => Ok(self.machine.our_value(true)),
+                Op::GreaterEqual => Ok(self.machine.our_value(lhs >= rhs)),
+                Op::LessEqual => Ok(self.machine.our_value(lhs <= rhs)),
```

Comment:
This again feels like it has little utility but I see that rust implements it. Maybe it's because there's a reasonable definition so why not? Maybe its to fulfill interfaces that might want ordering on units?

Reply:
As above, in Rust terms, I think you want `Foo<()>` to be ordered...

```rust
struct Foo<T> { v: u32, extra: T }
```